### PR TITLE
Fix missing translations when looking up language codes

### PR DESCRIPTION
**Pull request by hacodeorg**
[FND-1949](https://codedotorg.atlassian.net/browse/FND-1949): Fixing missing translations when opening a tutorial on http://hourofcode.com/learn and http://code.org/learn. 

What this PR does
- Move 2-letter and 3-letter language codes under a `language_code` key.
- Move locale code xx-yy under a `locale_code` key.
- Add missing language codes and locale codes from the pegaus.cdo_languages table.
- Update the `hoc_language` function, which maps codes to language names, to use both language codes and locale codes.
- Add unit tests for the `hoc_language` function.

## Links
- [Language codes reference (gsheet)](https://docs.google.com/spreadsheets/d/10KHKOKmpy34HIrCJkrLuv-Vy2acsm-Hz5KIDSV824VA/edit#gid=0)
- [cdo-tutorials (gsheet)](https://docs.google.com/spreadsheets/d/1Pw3Q_A-OpV33xbTvuWBXjEY10QmR-OvbiyT-Oxmce8U/edit#)

Previous PRs:
- https://github.com/code-dot-org/code-dot-org/pull/43096/
- https://github.com/code-dot-org/code-dot-org/pull/43466/

## Testing story
Unit tests
- `bundle exec ruby -Ilib:test pegasus/test/test_hourofcode_helpers.rb`

Manual tests
- Group language translations in `pegasus/sites.v3/hourofcode.com/i18n/es.yml` under `language_code` and `locale_code` keys.
- Test http://localhost.hourofcode.com:3000/learn in English, then Spanish (es-es).
- Test http://localhost.code.org:3000/learn in English, then Spanish (es-es).

## Follow up
- [x] Decide internally should we only use locale code xx-yy in [cdo-tutorials](https://docs.google.com/spreadsheets/u/1/d/1Pw3Q_A-OpV33xbTvuWBXjEY10QmR-OvbiyT-Oxmce8U/edit#). References are in [Language codes gsheet](https://docs.google.com/spreadsheets/d/10KHKOKmpy34HIrCJkrLuv-Vy2acsm-Hz5KIDSV824VA/edit#).
- [x] Ask translators to translate language names in Crowdin. Search for `language_code.*` and `locale_code.*`.

## PR Checklist:
- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

**Review by smusoke — APPROVED** [review]
Awesome!! This looks good to me

File: pegasus/helpers/hourofcode_helpers.rb [hunk 1]
```diff
@@ -110,12 +110,27 @@ def hoc_get_locale_code
 end
 
 # code.org and hourofcode.com's /learn pages call this to translate tutorial's languages attribute
-def hoc_language(lang_codes_str)
+def hoc_language(lang_codes_str, language = nil)
```
Comment (smusoke) [inline pegasus/helpers/hourofcode_helpers.rb:113]: Are we just adding the `language` parameter for testing?

File: pegasus/test/test_hourofcode_helpers.rb [hunk 2]
```diff
@@ -83,4 +84,55 @@ def test_redirect_untrusted_proxy
     response = get '/xyz'
     assert_equal 'http://hourofcode.com/us/xyz', response.headers['Location']
   end
+
+  def test_get_language_names_in_english
```
Comment (smusoke) [inline pegasus/test/test_hourofcode_helpers.rb:88]: Good test cases 👍  Thanks for adding these

File: pegasus/helpers/hourofcode_helpers.rb (hunk 1 above)
Comment (hacodeorg) [reply pegasus/helpers/hourofcode_helpers.rb:113]: The language param is only for testing, like dependency injection.  Without it we cannot call `hoc_s`, which relies on language data from request or url path.  We should design our functions in a way that is easier to test.

Comment (breville) [reply pegasus/helpers/hourofcode_helpers.rb:113]: Could we add some comments above these functions explaining it for future users?

**Pull request merged**